include/circt/Dialect/FIRRTL/FIRRTLAttributes.h

```diff
@@ -14,6 +14,110 @@
 #define CIRCT_DIALECT_FIRRTL_FIRRTLATTRIBUTES_H
 
 #include "circt/Dialect/FIRRTL/FIRRTLDialect.h"
+#include "circt/Dialect/FIRRTL/FIRRTLTypes.h"
+#include "circt/Support/LLVM.h"
+#include "llvm/ADT/BitVector.h"
+#include "llvm/ADT/iterator.h"
```

Comment:
This adds a fair amount of includes to a header. Are these necessary, can these be handled with forward declarations? If not, does that warrant a different factoring to put the implementation in a `*.cpp` file or as part of `FIRRTLOps.cpp`?

Reply:
I managed to drop a lot of the headers, but they are still included transitively.  I don't think there is anything we can move to the cpp file that will help, since they are needed for the class definition.